docs: update npm packages to be accurate for v7

jenniferfell opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ X] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Npm packages doc was not current for v7.
* Referred to old version of node
* Described yarn as default for CLI, not npm
* package lists were not current


Issue Number: N/A


## What is the new behavior?

The doc needs a larger rewrite, but this PR is to make it accurate in the ways mentioned above. 

Also made some copy edits. 

## Does this PR introduce a breaking change?
```
[ ] Yes
[X ] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

jenniferfell committed:
docs: update npm packages to be accurate for v7

IgorMinar commented on aio/content/guide/npm-packages.md:
```diff
@@ -37,17 +27,17 @@ You may even remove some.
 {@a dependencies}
 
-## *Dependencies*
+## Dependencies
```
it would be great if we could say that to add a new "Dependency" to the list, developers should use the "ng add" command.

jenniferfell replied:
done

IgorMinar commented on aio/content/guide/npm-packages.md:
```diff
@@ -113,11 +103,13 @@ which polyfills missing features for several popular browser.
 {@a dev-dependencies}
 
-## *DevDependencies*
+## DevDependencies
```
it would be great if we could say that to add a new "devDependency" to the list, developers should use the "npm install --dev" or "yarn add --dev" command.

jenniferfell replied:
done

IgorMinar commented on aio/content/guide/npm-packages.md:
```diff
@@ -155,19 +147,18 @@ Built on top of [WebDriverJS](https://github.com/SeleniumHQ/selenium/wiki/WebDri
 The browser downloads this bundle, not the original package files.
 
-See the [Deployment](guide/deployment) to learn more.
+See [Deployment](guide/deployment) to learn more.
```
instead we should point people to the build & serve docs to show them how all these packages come together to create a development (build & serve guide) and production (deployment guide) builds.

jenniferfell replied:
Done. Replaced "Managing packages and files" with a "Related information" section with links to build and deployment guides.

mary-poppins commented:
You can preview 5b38230 at https://pr26422-5b38230.ngbuilds.io/.

jenniferfell commented on aio/content/guide/npm-packages.md:
```diff
@@ -113,11 +103,13 @@ which polyfills missing features for several popular browser.
 The packages listed in the *devDependencies* section of the `package.json` help you develop the application on your local machine.
 
-You don't deploy them with the production application although there is no harm in doing so.
+You don't deploy them with the production application, although there is no harm in doing so.
```
super. done

jenniferfell commented on aio/content/guide/npm-packages.md:
```diff
@@ -113,11 +103,13 @@ which polyfills missing features for several popular browser.
 
```
good catch. thx. done

jenniferfell committed:
fixup! docs: update npm packages to be accurate for v7

mary-poppins commented:
You can preview 697ceab at https://pr26422-697ceab.ngbuilds.io/.

IgorMinar commented:
there are still many unresolved comments on this pr

IgorMinar commented on aio/content/guide/npm-packages.md:
```diff
@@ -113,11 +106,18 @@ which polyfills missing features for several popular browser.
+To add a new `devDependency`, use either one of the following commands, depending on which package manager you use:
 
-The packages listed in the *devDependencies* section of the `package.json` help you develop the application on your local machine.
+* `npm install --dev` 
```
`npm install --dev <name-of-the-package>`

jenniferfell replied:
Done

IgorMinar commented on aio/content/guide/npm-packages.md:
```diff
@@ -113,11 +106,18 @@ which polyfills missing features for several popular browser.
 
-The packages listed in the *devDependencies* section of the `package.json` help you develop the application on your local machine.
+* `npm install --dev` 
+* `yarn add --dev` 
```
add `<name-of-the-package>`

jenniferfell replied:
Done

mary-poppins commented:
You can preview 4e3b615 at https://pr26422-4e3b615.ngbuilds.io/.

mary-poppins commented:
You can preview 2de0e55 at https://pr26422-2de0e55.ngbuilds.io/.

jenniferfell commented on aio/content/guide/npm-packages.md:
```diff
@@ -58,19 +48,21 @@ The `dependencies` section of `package.json` contains:
 
 ### Angular Packages
 
+For a complete list of Angular packages (and links to package overviews), see the [API reference](http://angular.io/api?type=package). The following packages are included as dependencies in the initial `package.json` file. 
```
done

jenniferfell commented on aio/content/guide/npm-packages.md:
```diff
@@ -37,17 +27,17 @@ You may even remove some.
 
 This guide focuses on the most important packages in the starter set.
```
done and thanks for the link

jenniferfell committed:
fixup! docs: update npm packages to be accurate for v7

jenniferfell commented on aio/content/guide/npm-packages.md:
```diff
@@ -1,30 +1,20 @@
+
+## `package.json`
 
 Both `npm` and `yarn` install packages that are identified in a [**package.json**](https://docs.npmjs.com/files/package.json) file.
```
Added note about yarn back in above

jenniferfell commented on aio/content/guide/npm-packages.md:
```diff
@@ -1,30 +1,20 @@
-You can download and install these npm packages with the [**npm client**](https://docs.npmjs.com/cli/install), which runs as a Node.js® application.
+You can download and install these npm packages by using the [**npm CLI client**](https://docs.npmjs.com/cli/install), which is installed with and runs as a Node.js® application. By default, the Angular CLI uses the npm client to install npm packages when you create a new project.
 
-The [**yarn client**](https://yarnpkg.com/en/) is a popular alternative for downloading and installing npm packages.
```
Done.

mary-poppins commented:
You can preview b668c2d at https://pr26422-b668c2d.ngbuilds.io/.

jenniferfell commented:
@IgorMinar : Ready for another look. 

Q1: is package.json  workspace-scope  or project-scope? I see that ng new creates a package.json at the workspace level. I see that ng generate app does not create a package.json in my new app project under projects.

Current text: ”The CLI `ng new` command creates a default `package.json` file for your project.”

Seems like it should either be that `ng new` creates a `package.json` for your workspace, or that both `ng new` and `ng generate app` create `package.json` when they create projects?

* Q2: Page title “Angular packages and dependencies” and nav title “Packages and dependencies” ? OR “Angular package dependencies” and “Package dependencies” ? Just trying something other than “npm packages” if you have a quick pref to change.

* Q3: Are these the best URLs for the devDependencies that begin @angular? 

Thanks!

IgorMinar commented on aio/content/guide/npm-packages.md:
```diff
@@ -86,47 +81,53 @@ using the [JIT compiler](guide/aot-compiler).
-
+**[@angular/compiler-cli](https://github.com/angular/angular/blob/master/packages/compiler-cli/README.md)**: The Angular compiler, which is invoked by the Angular CLI's `ng build` and `ng serve` commands.
 
 **[@angular/language-service](https://github.com/angular/angular-cli/)**: The Angular language service analyzes component templates and provides type and error information that TypeScript-aware editors can use to improve the developer's experience.
```
I don't know why this points to the cli repo, but that's wrong. please remove the link.

jenniferfell replied:
Done

IgorMinar reviewed (approved):
A1: workspace scoped
A2:  neither is great. packages are dependencies. so we just repeat the same thing using two different words. how about: “npm dependencies” and “Workspace npm dependencies”?
A3: left in inline comment. the rest looks good.

IgorMinar commented:
I'm going to approve this because it's almost done. Let's change the title, resolve the other issues and get this in.

mary-poppins commented:
You can preview 5791c46 at https://pr26422-5791c46.ngbuilds.io/.

jenniferfell committed:
fixup! docs: update npm packages to be accurate for v7

Splaktar commented on aio/content/guide/npm-packages.md:
```diff
@@ -1,76 +1,71 @@
+
+Both `npm` and `yarn` install packages that are identified in a [`package.json`](https://docs.npmjs.com/files/package.json) file.
 
 The CLI `ng new` command creates a default `package.json` file for your project.
```
`project` should actually be `workspace` here.

jenniferfell replied:
Done

Splaktar commented:
> Q1: is package.json workspace-scope or project-scope? I see that ng new creates a package.json at the workspace level. I see that ng generate app does not create a package.json in my new app project under projects.

> Current text: ”The CLI ng new command creates a default package.json file for your project.”

> Seems like it should either be that ng new creates a package.json for your workspace, or that both ng new and ng generate app create package.json when they create projects?

Right, as mentioned by Igor, `package.json` in the root of the workspace is workspace-scope. `ng g app` doesn't create a `package.json` for new projects added to the workspace in `projects/`.

I added a comment on the line that "project" should be "workspace" in this case.

Note that `ng g library` does however create a `package.json` for the library, but it's really there for publishing the library to NPM. It's not something that would be `npm install`ed in the project dir. I'm sure that the library details are covered elsewhere in the docs and doesn't need to be part of this page.

mary-poppins commented:
You can preview d1eb8bd at https://pr26422-d1eb8bd.ngbuilds.io/.

jenniferfell committed:
fixup! docs: update npm packages to be accurate for v7

jenniferfell commented:
> Note that `ng g library` does however create a `package.json` for the library, but it's really there for publishing the library to NPM. It's not something that would be `npm install`ed in the project dir. I'm sure that the library details are covered elsewhere in the docs and doesn't need to be part of this page.

Thanks for this tip. I don't think it's covered in the AIO docs, and I don't see it clearly stated on the CLI wiki. In general, we're trying to do a better job of including information about library development (we have a few top-of-backlog items), so I included a brief mention and link to `ng generate` doc. We'll need to add more to that doc and address the backlog need for guide material, but I do think this is the right place to mention that libs have package.json files. Thanks again!

jenniferfell committed:
fixup! docs: update npm packages to be accurate for v7

jenniferfell committed:
fixup! docs: update npm packages to be accurate for v7

mary-poppins commented:
You can preview 296f04f at https://pr26422-296f04f.ngbuilds.io/.

jenniferfell committed:
fixup! docs: update npm packages to be accurate for v7

mary-poppins commented:
You can preview a368f7c at https://pr26422-a368f7c.ngbuilds.io/.

jenniferfell commented:
@IgorMinar Here, again. I changed navigation.json so now the PR needs a fresh approval from someone in the marketing group. Thx.

jenniferfell commented:
Other than that approval, I think it's ready. I've incorporated all review feedback, updated and added cross-reference links, and moved the lists into tables that are easier to read. (When I added links, scanning the in-line titles became harder. The table fixes that.)

Splaktar reviewed (approved):
LGTM

mary-poppins commented:
You can preview d9415fb at https://pr26422-d9415fb.ngbuilds.io/.
